codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonMemberDeserVisitor.java

```diff
@@ -0,0 +1,349 @@
+import software.amazon.smithy.model.shapes.IntegerShape;
+import software.amazon.smithy.model.shapes.ListShape;
+import software.amazon.smithy.model.shapes.LongShape;
+import software.amazon.smithy.model.shapes.MapShape;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ResourceShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.SetShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeVisitor;
+import software.amazon.smithy.model.shapes.ShortShape;
+import software.amazon.smithy.model.shapes.StringShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.shapes.TimestampShape;
+import software.amazon.smithy.model.shapes.UnionShape;
+import software.amazon.smithy.model.traits.EnumTrait;
+import software.amazon.smithy.model.traits.TimestampFormatTrait.Format;
+
+/**
+ * Visitor to generate member values for aggregate types deserialized from documents.
+ */
+public class JsonMemberDeserVisitor implements ShapeVisitor<Void> {
+    private final GenerationContext context;
+    private final String dataDest;
+    private final Format timestampFormat;
+
+    public JsonMemberDeserVisitor(GenerationContext context, String dataDest, Format timestampFormat) {
+        this.context = context;
+        this.dataDest = dataDest;
+        this.timestampFormat = timestampFormat;
+    }
+
+    @Override
+    public Void blobShape(BlobShape shape) {
+        GoWriter writer = context.getWriter();
+        writer.write("err := decoder.Decode(&$L)", dataDest);
+        writer.write("if err != nil { return err }");
+        return null;
+    }
+
+    @Override
+    public Void booleanShape(BooleanShape shape) {
+        GoWriter writer = context.getWriter();
+        writer.addUseImports(SmithyGoDependency.FMT);
+        consumeToken();
+        writer.openBlock("if val != nil {", "}", () -> {
+            writer.write("jtv, ok := val.(bool)");
+            writer.openBlock("if !ok {", "}", () -> {
+                writer.write("return fmt.Errorf(\"expected $L to be of type *bool, got %T instead\", val)",
+                        shape.getId().getName());
+            });
+            writer.write("$L = &jtv", dataDest);
+        });
+        return null;
+    }
+
+    /**
+     * Consumes a single token into the variable "val", returning on any error.
+     */
+    private void consumeToken() {
+        GoWriter writer = context.getWriter();
+        writer.write("val, err := decoder.Token()");
+        writer.write("if err != nil { return err }");
+    }
+
+    @Override
+    public Void byteShape(ByteShape shape) {
+        GoWriter writer = context.getWriter();
+        handleInteger(shape, CodegenUtils.generatePointerValueIfPointable(writer, shape, "int8(i64)"));
+        return null;
+    }
```

Comment:
minor comment might be helpful callout out that a Smithy Byte doesn't map to Go's byte int8 vs uint8.